MAUI/DataGrid/columns.md

```diff
@@ -0,0 +1,205 @@
+
+Below table shows the column type created for the respective data types. For remaining types, [DataGridTextColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridTextColumn.html) will be created.
+
+<table>
+<tr>
+<th> Data Type </th>
+<th> Column </th>
+</tr>
+<tr>
+<td> string, object </td>
+<td> DataGridTextColumn </td>
+</tr>
+<tr>
+<td> int, float, double, decimal and it’s respective nullable types </td>
+<td> DataGridNumericColumn </td>
+</tr>
+<tr>
+<td> DateTime </td>
+<td> DataGridDateColumn </td>
+</tr>
+<tr>
+<td> bool </td>
+<td> DataGridCheckboxColumn </td>
+</tr>
+<tr>
+<td> ImageSource </td>
+<td> DataGridImageColumn </td>
+</tr>
+</table>
+
+### Different modes to auto generate columns
+
+The auto generation of the columns in `SfDataGrid` is happening based on the `SfDataGrid.AutoGenerateColumnsMode` property. The default value is [AutoGenerateColumnsMode.Reset](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.AutoGenerateColumnsMode.html#Syncfusion_Maui_DataGrid_AutoGenerateColumnsMode_Reset).
+The `SfDataGrid.AutoGenerateColumnsMode` includes the following modes,
+
+<table>
+<tr>
+<th> Modes </th>
+<th> Description </th>
+</tr>
+<tr>
+<td> <code>None</code> </td>
+<td> Stores only the columns that are defined in <code>SfDataGrid.Columns</code> collection.<br/> When changing the <code>ItemsSource</code> and sorting for explicitly defined <code>SfDataGrid.Columns</code> alone will be retained. </td>
+</tr>
+<tr>
+<td> <code>Reset</code> </td>
+<td> Retains the columns defined explicitly in the application level and creates columns newly for all the other properties in a data source.<br/> When changing the <code>ItemsSource</code> and sorting for explicitly defined <code>SfDataGrid.Columns</code> alone will be retained. </td>
+</tr>
+<tr>
+<td> <code>ResetAll</code> </td>
+<td> When changing the <code>ItemsSource</code>, the columns for the previous data source are cleared and the columns will be created newly for the new data source. Even when columns are explicitly defined it does not consider the defined columns and creates the column based on the underlying collection.<br/> Further when changing the <code>ItemsSource</code> and sorting for all the columns will be cleared. </td>
+</tr>
+<tr>
+<td> <code>RetainOld</code> </td>
+<td> When changing the <code>ItemsSource</code>, creates columns for all fields in a data source when the DataGrid does not have any explicit definition for columns. When columns are defined explicitly, then the defined columns alone are retained and new columns are not created.<br/>Similarly when changing the <code>ItemsSource</code> and when the DataGrid have any explicit definition for columns and sorting are retained as it is. </td>

```

Comment:
<td> When changing the <code>ItemsSource</code>, creates columns for all fields in a data source when the DataGrid does not have any explicit definition for columns. When columns are defined explicitly, then the defined columns alone are retained and new columns are not created.<br/>Similarly, when changing the <code>ItemsSource</code> and when the DataGrid has an explicit definition for columns and sorting are retained as it is. </td>

Reply:
Modified.